js/base-chat.js

```diff
@@ -0,0 +1,617 @@
+				},
+			},
+		});
+	}
+
+	const languageTemplate = () => `
+		<script id="sheet-rolltemplate-inlanguage" type="text/html">{{displaymessage}}<br>
+			<span style="font-style:italic" title="You understand this because one of your characters speaks {{titlelanguage}}">
+			<span style="font-weight:bold">{{displaylanguage}}</span> {{translated}}
+			</span>
+		</script>
+	`;
+
+	const languageDialogTemplate = (msg, language) => `
+			<div>
+				<p><strong>What does ${language} language sound like?</strong></p>
+				<p>It seems you're trying to speak language not included in the standard list of D&D PHB.</p>
+				<p>That's not a problem. Please select one of the languages from the dropdown below, and it will be used for the imitated message.</p>
+				<p>Your choice is purely cosmetic and will not affect who can or can not understand it. This will be remembered until you refresh the page.</p>
+				<select id="soundslike">
+					${Object.keys(languages).reduce((options, lang) => `${options}<option value="${lang}">${lang}</option>`, "")}
+				</select>
+				<p><i>Example: <span id="languageeg">${gibberish(msg, "common")}</span></i></p>
+			</div>
+	`;
+
+	const playerConnectsTemplate = (id) => `
+			<input type="checkbox" class="connects-state" id="connects${id}-state"/>
+			<label for="connects${id}-state">
+				<span id="connects${id}-info" class="connects-info" title="Show player details">?</span>
+				<span id="connects${id}" class="connects-log">
+				Updating...
+				</span>
+			</label>
+	`;
+
+	const chatHelp = [
+		["/w (name)", "private message (whisper)"],
+		["/w gm", "private message to your GM"],
+		["/wb", "reply or whisper to last contact", "b20"],
+		["/ws", "whisper to selected tokens", "b20"],
+		["/em, /me", "emote (action from your POV)"],
+		["/in (language)", "speak in a language", "b20"],
+		["--(word)", "skip word (for in-language)", "b20"],
+		// ["/cl on|off", "comprehend language switch", "b20"],
+		["/talktomyself", "silent mode on/off"],
+		["/ttms", "shortcut to /talktomyself", "b20"],
+		["/mtms (text)", "simple message to self", "b20"],
+		["/ooc, /o", "out of character (as player)"],
+		["/roll, /r (XdY)", "roll dice, e.g. /r 2d6"],
+		["/gmroll, /gr (XdY)", "roll only for GM"],
+		["/desc", "GM-only describe events", "gm"],
+		["/as (name)", "GM-only speak as Name", "gm"],
+		["/emas (name)", "GM-only emote as Name", "gm"],
+		["&#42;(text)&#42;", "format text: italic"],
+		["&#42;&#42;(text)&#42;&#42;", "format text: bold"],
+		["&#126;&#126;(text)&#126;&#126;", "format text: strikethrough"],
+		["&#96;&#96;(text)&#96;&#96;", "format text: code"],
+		["/fx (params)", "show visual effect"],
+		["#(macro)", "run specified macro"],
+		["/help", "show this list of chat commands", "b20"],
+		["", "<a style=\"font-variant: diagonal-fractions;font-size: smaller;font-variant-caps: small-caps;\" href=\"https://wiki.roll20.net/Text_Chat\">roll20 wiki</a>"],
+	];
+
+	d20plus.chat.help = (text, msg) => {
```

Comment:
Can you make this look better on light mode? You're allowed to say you don't support light mode because you refuse to test on it though.

Reply:
Done, I think. Now it looks like vttes welcome msg